modules/drivers/snowflake/src/metabase/driver/snowflake.clj

```diff
@@ -42,16 +43,13 @@
   :sunday)
 
 (defmethod sql-jdbc.conn/connection-details->spec :snowflake
-  [_ {:keys [account regionid], :as opts}]
-  (let [host (if regionid
-               (str account "." regionid)
-               account)
-        upcase-not-nil (fn [s] (when s (u/upper-case-en s)))]
+  [_ {:keys [account], :as opts}]
```

Comment:
why aren't you doing the normalizing here. What's the point of having the separate defmethod rather than doing it here?

Reply:
This was at Cam's initial suggestion, but happy to go either way.

Reply:
Fair enough. Roll with it

Reply:
@dpsutton, that's a totally valid point. 

I think the only situation where adding a new `normalize-db-details` method would be useful would be if we needed the transformation to apply to other places the details are used as well. AFAIK the only other place we use DB details is in the Edit Database page in the admin panel. 

In this case it looks like we actually want to append region id to `:account` itself, and that _should_ be something visible in the admin Edit Database page, so the `normalize-db-details` method seems appropriate.

If I'm reading the code wrong and the changes don't actually need to affect the admin page then I agree with Dan, we should just do the normalization in `connection-details->spec` and skip adding the new method

Reply:
Yeah, so in this case, the changes should definitely affect the admin page, **and** other pages.

- When we render the DB edit page, the `regionid` field is simply gone, so its value will need to have been migrated over to `account`, so that it will be appended there on save.
- Even if the user never actually saves it again, the `:regionid` is still gone, and the connection functions are no longer trying to find it in the details map.  Thus, to maintain consistency between what the connectivity code uses, and what the user might see on the admin screen, it should still be "normalized" even for connections.

Given this, I suppose the conclusion stands to have the multimethod.